scm/scm_options.go

```diff
@@ -0,0 +1,151 @@
+package scm
+
+import "github.com/quickfeed/quickfeed/qf"
+
+// CourseOptions contain information about new course.
+type CourseOptions struct {
+	OrganizationID uint64
+	CourseCreator  string
+}
+
+func (opt CourseOptions) valid() bool {
+	return opt.OrganizationID > 0 && opt.CourseCreator != ""
+}
+
+// GroupOptions contain information about group.
+type GroupOptions struct {
+	OrganizationID uint64
+	RepositoryID   uint64
+	TeamID         uint64
+}
+
+func (opt *GroupOptions) valid() bool {
+	return opt.OrganizationID > 0 && opt.RepositoryID > 0 &&
+		opt.TeamID > 0
+}
+
+// UpdateEnrollmentOptions contain information about enrollment.
+type UpdateEnrollmentOptions struct {
+	Organization string
+	User         string
+	Status       qf.Enrollment_UserStatus
+}
+
+func (opt UpdateEnrollmentOptions) valid() bool {
+	return opt.Organization != "" && opt.User != ""
+}
+
+// RejectEnrollmentOptions contain information about enrollment.
+type RejectEnrollmentOptions struct {
+	OrganizationID uint64
+	RepositoryID   uint64
+	User           string
+}
+
+func (opt *RejectEnrollmentOptions) valid() bool {
+	return opt.OrganizationID > 0 && opt.RepositoryID > 0 &&
+		opt.User != ""
+}
+
+// OrganizationOptions contain information about organization.
+type OrganizationOptions struct {
+	ID   uint64
+	Name string
+	// Username field is used to filter organizations
+	// where the given user has a certain role.
+	Username  string
+	NewCourse bool // Get organization for a new course
+}
+
+func (opt OrganizationOptions) valid() bool {
+	return opt.ID != 0 || opt.Name != ""
+}
+
+// RepositoryOptions is used to fetch a single repository by ID or name.
+// Either ID or both Path and Owner fields must be set.
+type RepositoryOptions struct {
+	ID    uint64
```

Comment:
Should we use `RepositoryID` to avoid confusion with other IDs?

Reply:
On second thought. Let's keep these as they are to avoid further polluting this PR with unrelated changes.